plugins/inputs/mysql/mysql.go

```diff
@@ -215,25 +657,677 @@ func (m *Mysql) gatherServer(serv string, acc telegraf.Accumulator) error {
+	rows, err := db.Query(infoSchemaAutoIncQuery)
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+
+	var (
+		schema, table, column string
+		incValue, maxInt      uint64
+	)
+
+	servtag, err := parseDSN(serv)
+	if err != nil {
+		servtag = "localhost"
+	}
+
+	for rows.Next() {
+		if err := rows.Scan(&schema, &table, &column, &incValue, &maxInt); err != nil {
+			return err
+		}
+		tags := map[string]string{
+			"server": servtag,
+			"schema": schema,
+			"table":  table,
+			"column": column,
+		}
+		fields := make(map[string]interface{})
+		fields["auto_increment_column"] = incValue
+		fields["auto_increment_column_max"] = maxInt
+
+		acc.AddFields("mysql_info_schema", fields, tags)
+	}
+	return nil
+}
+
+// gatherPerfTableLockWaits can be used to get
+// the total number and time for SQL and external lock wait events
+// for each table and operation
+func (m *Mysql) gatherPerfTableLockWaits(db *sql.DB, serv string, acc telegraf.Accumulator) error {
+	rows, err := db.Query(perfTableLockWaitsQuery)
+	if err != nil {
+		return err
+	}
+	defer rows.Close()
+
+	servtag, err := parseDSN(serv)
+	if err != nil {
+		servtag = "localhost"
+	}
+
+	var (
+		objectSchema               string
+		objectName                 string
+		countReadNormal            uint64
+		countReadWithSharedLocks   uint64
+		countReadHighPriority      uint64
+		countReadNoInsert          uint64
+		countReadExternal          uint64
+		countWriteAllowWrite       uint64
+		countWriteConcurrentInsert uint64
+		countWriteDelayed          uint64
+		countWriteLowPriority      uint64
+		countWriteNormal           uint64
+		countWriteExternal         uint64
+		timeReadNormal             uint64
+		timeReadWithSharedLocks    uint64
+		timeReadHighPriority       uint64
+		timeReadNoInsert           uint64
+		timeReadExternal           uint64
+		timeWriteAllowWrite        uint64
+		timeWriteConcurrentInsert  uint64
+		timeWriteDelayed           uint64
+		timeWriteLowPriority       uint64
+		timeWriteNormal            uint64
+		timeWriteExternal          uint64
+	)
+
+	for rows.Next() {
+		err = rows.Scan(
+			&objectSchema,
+			&objectName,
+			&countReadNormal,
+			&countReadWithSharedLocks,
+			&countReadHighPriority,
+			&countReadNoInsert,
+			&countReadExternal,
+			&countWriteAllowWrite,
+			&countWriteConcurrentInsert,
+			&countWriteDelayed,
+			&countWriteLowPriority,
+			&countWriteNormal,
+			&countWriteExternal,
+			&timeReadNormal,
+			&timeReadWithSharedLocks,
+			&timeReadHighPriority,
+			&timeReadNoInsert,
+			&timeReadExternal,
+			&timeWriteAllowWrite,
+			&timeWriteConcurrentInsert,
+			&timeWriteDelayed,
+			&timeWriteLowPriority,
+			&timeWriteNormal,
+			&timeWriteExternal,
+		)
+
+		if err != nil {
+			return err
+		}
+		tags := map[string]string{
+			"server": servtag,
+			"schema": objectSchema,
+			"table":  objectName,
+		}
+		fields := make(map[string]interface{})
+
+		tags["operation"] = "read_normal"
```

Comment:
@Maksadbek you should not be mutating tag maps after calling AddFields, see https://github.com/influxdata/telegraf/blob/master/accumulator.go#L9

so you'll need to refactor all code that is passing a `tag` or `fields` map and then mutating that map afterwards.